### DISCUSSION TITLE
How to Create a Top Bar which is Exclusive

### DISCUSSION THREAD

**Discussion by Kynson** [elkowar/eww#1049] [Q&A] [answered]
Hello, I am trying to create a top bar positioned at the very top of my screen but I get stuck on how to configure its position.

What I am trying to achieve:
- A simple horizontal bar positioned at the very top of the screen which cannot be overlapped/ overlayed by other windows (i.e. exclusive?)

I tried to make the bar exclusive by setting exclusive to true in my config and set the anchor to `left center` as stated in #296. Yet, I now lost control to the y-position (vertical), no matter what value I set to the window's `geometry.y` property it just stays at the center.

I wonder how can I configure the bar so that it is positioned at the top and it can be marked as exclusive? Currently it seems to be an either-or situation: either the bar is at the top but exclusive is not working, or the bar is at the center and preventing any other window to be launched (as it is marked as exclusive, I can't launch new windows that overlap with it).

Window Manager: Hyprland
eww version: 0.4.0 (eww-wayland)
OS: NixOS 23.11

eww.yuck:

```
(defwidget topBar []
    (box
        :orientaion "horizontal"
        :spacing 20
        "Test top bar"
        (button :onclick "poweroff" "Shut Down")
     )
)
(defwindow topBar
    :monitor 0
    geometry (geometry
        ;; Does not work?
        :y "4%"
        :width "1248px"
        :height "4%"
        :anchor "left center"
        :exclusive true
        :focusable true
    )
    (topBar)
)
```

I am very new to this field and any suggestion is welcome. Thanks in advance!

**Comment by Rayzeq** [comment, marked as answer]
The `exclusive` and `focusable` need to be outside of the `geometry` node.
According to [this issue](https://github.com/elkowar/eww/issues/296#issuecomment-1152982035) your anchor is also invalid, you should use `top center`.
For some reason there is also a colon missing before `geometry`.
Here's the fixed config:
```
(defwindow topBar
    :monitor 0
    :geometry (geometry
        :y "4%"
        :width "1248px"
        :height "4%"
        :anchor "top center"
    )
    :exclusive true
    :focusable true
    (topBar)
)

**Reply by Kynson** [reply]
Thanks! It is now working. The syntax errors were caused when I was formatting the configuration in GitHub, sorry for the confusion. The `top center` anchor does the trick!
